Makefile.toml

```
@@ -11,267 +10,483 @@ skip_core_tasks = true
 skip_git_env_info = true
 skip_rust_env_info = true
 
-
 [tasks.default]
-description = "List tasks"
+description = "List available tasks"
+category = "Default"
 script = "cargo make --quiet --list-all-steps --hide-uninteresting"
 
-
-# Build
+# =================================================================================================
+# Development Flows - Quick iteration for local development
+# =================================================================================================
+[tasks.dev]
+description = "Development flow with all features"
+category = "Development"
+run_task = { name = ["format", "build", "clippy"] }
+
+[tasks.dev-default]
+description = "Development flow with default features"
+category = "Development"
+run_task = { name = ["format", "build-default", "clippy"] }
+
+[tasks.dev-preview]
+description = "Development flow with preview features"
+category = "Development"
+run_task = { name = ["format", "build-preview", "clippy"] }
+
+[tasks.dev-experimental]
+description = "Development flow with experimental features"
+category = "Development"
+run_task = { name = ["format", "build-experimental", "clippy"] }
+
+# =================================================================================================
+# Quality Assurance - Full checks before committing
+# =================================================================================================
+[tasks.qa]
+description = "Quality assurance checks"
+category = "Quality"
+run_task = { name = ["check-format", "clippy", "check-features", "security-flow", "test", "example-flow"] }
+
+# =================================================================================================
+# CI Flows - Comprehensive testing for continuous integration
+# =================================================================================================
+[tasks.ci]
+description = "Complete CI - all checks that run in GitHub Actions"
+category = "CI"
+run_task = { name = ["ci-correctness", "ci-tests", "ci-build", "ci-docs", "ci-security"] }
+
+[tasks.ci-default]
+description = "Complete CI with default features"
+category = "CI"
+run_task = { name = ["ci-correctness", "ci-tests", "ci-build-default", "ci-docs", "ci-security"] }
+
+[tasks.ci-preview]
+description = "Complete CI with preview features"
+category = "CI"
+run_task = { name = ["ci-correctness", "ci-tests", "ci-build-preview", "ci-docs", "ci-security"] }
+
+[tasks.ci-experimental]
+description = "Complete CI with experimental features"
+category = "CI"
+run_task = { name = ["ci-correctness", "ci-tests", "ci-build-experimental", "ci-docs", "ci-security"] }
+
+[tasks.ci-correctness]
```

We can keep `correctness` as the name of this task. The idea of `cargo make` is that the same tasks can be invoked by developers and/or CI. I don't think we want to tie CI down to using particular tasks in `Makefile.toml`. I'd rather let CI and developers select whichever tasks they need to. The `.yml` workflows are implemented at a level above `Makefile.toml` in the architecture, so we shouldn't reference `correctness.yml` or CI at all here.
```suggestion
[tasks.correctness]
```